### ISSUE TITLE
Unable to Run imagick on Windows Client for J2K Image Format Conversion

### ISSUE THREAD

**Issue by Zev3691** [gographics/imagick#302]
errmsg: NoDecodeDelegateForThisImageFormat `J2K' @ error/blob.c/BlobToImage/462

I successfully resolved the issue related to running the program on Windows clients (refer to [Issue #211](https://github.com/gographics/imagick/issues/211)), but now I am facing difficulties running imagick for J2K image format conversion on client machines.

When attempting to convert J2K images using imagick, the application fails to run on client machines. I am uncertain about the necessary steps to make it work. Do I need to copy specific .dll files to the client machine and add them to the environment variables? Or are there alternative methods that do not require clients to install additional dependencies?

**Comment by justinfx** [comment]
The underlying imagemagick C library is looking for a dynamic dependency that provides the J2K delegate plugin. So it must exist on the client machine.

**Comment by justinfx** [comment]
Also, I don't have much experience compiling on windows. But in a linux environment once could static link everything in the ImageMagick library build so that it doesn't need external delegate shared libraries. Seems people have asked for similar guidance under Windows, to accomplish static builds:
https://www.imagemagick.org/discourse-server/viewtopic.php?t=12032

**Comment by Zev3691** [comment]
Thank you for your prompt response! I've seemingly found a solution to the issue. 
Following the instructions in the markdown file, 
I discovered that on Windows, the `PKG_CONFIG_PATH` and `MAGICK_CODER_MODULE_PATH` need to be set as custom environment variables. 
I moved them out of the `PATH` as described in the markdown. 
After executing `go build` again, the generated .exe file now works as expected on client machines.


I'm not sure if this method is universally applicable, but it successfully allowed my program to execute as intended.
